### PR TITLE
feat(argo-rollouts): add dashboard permission on Deployments

### PR DESCRIPTION
**Pull request by bpoland**
- needed for Rollouts with workload referencing

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

File: charts/argo-rollouts/templates/dashboard/clusterrole.yaml [hunk 1]
```diff
@@ -43,6 +43,7 @@ rules:
   - apiGroups:
       - apps
     resources:
+      - deployments
```
Comment (huikang) [inline charts/argo-rollouts/templates/dashboard/clusterrole.yaml:46]: Looks good to me. How about making this consistent to the cluster role as in the argo-rollouts repo:

https://github.com/argoproj/argo-rollouts/blob/55a041a6b3d088cf9ddcfc4121ad8c97880489db/manifests/role/argo-rollouts-clusterrole.yaml#L60-L69

Comment (bpoland) [reply charts/argo-rollouts/templates/dashboard/clusterrole.yaml:46]: Sure will do!

**Review by huikang — APPROVED** [review]

**Comment by bpoland** [comment]
@huikang do you know how to get the `chart-test` workflow to run, and get this PR merged?  Thanks!

**Comment by huikang** [comment]
> @huikang do you know how to get the `chart-test` workflow to run, and get this PR merged? Thanks!

I guess it needs any of maintainer's review to trigger the build.

**Comment by bpoland** [comment]
@mbevc1 it looks like you have approved some other PRs recently, would you mind taking a look at this one too?  Thanks :)

File: charts/argo-rollouts/templates/dashboard/clusterrole.yaml [hunk 2]
```diff
@@ -40,6 +40,17 @@ rules:
       - get
       - list
       - watch
+  # deployments and podtemplates read access needed for workload reference support
+  - apiGroups:
+      - ""
```
Comment (mbevc1) [inline charts/argo-rollouts/templates/dashboard/clusterrole.yaml:45]: Is this intentional?

Comment (bpoland) [reply charts/argo-rollouts/templates/dashboard/clusterrole.yaml:45]: I was copying from here as @huikang recommended: https://github.com/argoproj/argo-rollouts/blob/55a041a6b3d088cf9ddcfc4121ad8c97880489db/manifests/role/argo-rollouts-clusterrole.yaml#L60-L69

Comment (bpoland) [reply charts/argo-rollouts/templates/dashboard/clusterrole.yaml:45]: I think it's because `PodTemplate` doesn't have an API group?

Comment (mbevc1) [reply charts/argo-rollouts/templates/dashboard/clusterrole.yaml:45]: Ah, I see :+1:

**Review by mbevc1 — APPROVED** [review]

**Comment by bpoland** [comment]
Thanks!

**Comment by mbevc1** [comment]
No worries, thanks for your contribution.

**Pull request merged**